portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,31 +36,33 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> greetings = new ArrayList<String>();
-    String json = convertToJsonUsingGson(greetings);
+    Query query = new Query("comment");
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList messages = new ArrayList<>();
+    for (Entity entity : results.asIterable()){
+        long id = entity.getKey().getId();
+        String msg = (String) entity.getProperty("comment");
```

Comment:
Consider making constants like strings (and integers, etc.) as a static class member variable (either public or private depending on whether other classes may need to access them). For example:

    (public|private) static final String COMMENT = "comment";

The reason behind is there is a single source of truth and it's easy to make future changes (say if you need to change it to "comments") and avoid missing updating some, which is usually bug-prone.

Reply:
Forgot to mention, in Java, constant naming convention usually capitalize all letters.

Reply:
So in this file would that be query, json, text, msgEntity, but msg wouldn't be since it gets updated each iteration?

Reply:
Oops, I wasn't clear enough. By saying constants, I meant specifically the "comment" string in your case here. In general, they are things whose content remain the same across the whole server lifecycle. For example, the content of json will change, so that's not constant so you are fine with it.

Another trivial example that the professor gave me in lectures. Think of Pi when you think of constants. You would do "public final double PI = 3.14". Hope it's clear now.